kernel/syscall.c

```diff
@@ -3169,6 +3169,20 @@ long myst_syscall_mbind(
     return ret;
 }
 
+long myst_syscall_get_mempolicy(
+    int* mode,
+    unsigned long* nodemask,
+    unsigned long maxnode,
+    void* addr,
+    unsigned long flags)
+{
+    long params[6] = {
+        (long)mode, (long)nodemask, (long)maxnode, (long)addr, (long)flags};
+    long ret = myst_tcall(SYS_get_mempolicy, params);
+
```

Comment:
From Linux Manual:

       If flags specifies MPOL_F_MEMS_ALLOWED (available since Linux
       2.6.24), the mode argument is ignored and the set of nodes
       (memories) that the thread is allowed to specify in subsequent
       calls to mbind(2) or set_mempolicy(2) (in the absence of any mode
       flags) is returned in nodemask.

As Mystikos does not support mbind or set_mempolicy, it seems that the get_mempolicy() implementation should override nodemask returned by the target to indicate the thread is not allow to specify policy on any node in mbind or set_mempolicy.

Reply:
Added the overriding - this creates an assert failure in the `make native` target though that I have left there with a comment of explanation, since both are right in what they do